src/shared/Shared_List.re

```diff
@@ -1,15 +1,8 @@
-// TODO: pretty sure there are better ways to implement this
-let split = (~delimiter, lst) => {
-  let rec go = (accList, acc, delimiter, lst) =>
-    switch(lst) {
-    | [] =>
-      [acc, ...accList]
-      |> Relude.List.reverse
-    | [x, ...xs] =>
-      x == delimiter
-      ? go([acc, ...accList], [], delimiter, xs)
-      : go(accList, [x, ...acc], delimiter, xs)
-    };
-  go([], [], delimiter, lst)
-  |> Relude.List.map(Relude.List.reverse)
-};
\ No newline at end of file
+let split = (~delimiter) =>
+  Relude.List.foldl([[]], (b, a) =>
+    a == delimiter
+      ? [...b, []]
```

Comment:
Won't be able to use a list like this in Reason I think
```
Error: Lists can only have one `...` spread, at the end.
Explanation: lists are singly-linked list, where a node contains a value and
points to the next node. `[a, ...bc]` efficiently creates a new item and
links `bc` as its next nodes. `[...bc, a]` would be expensive, as it'd need
to traverse `bc` and prepend each item to `a` one by one. We therefore
disallow such syntax sugar.
Solution: directly use `concat` or other List
helpers.
```

Reply:
I just fixed it, i did the first edit in the github web editor lol :blush:

Reply:
Nice, thanks!

Wonder if it might be better to use an array type for the work since appending to lists is so inefficient, but it's not enough to even notice when I tested with this version so not worrying about it too much for now.